src/stream/src/executor/dedup/append_only_dedup.rs

```diff
@@ -84,15 +86,19 @@ impl<S: StateStore> AppendOnlyDedupExecutor<S> {
 
                     // Ensure that if a key for a visible row exists before, then it is in the
                     // cache, by querying the storage.
-                    self.populate_cache(keys.iter().flatten()).await?;
+                    self.populate_cache(dedup_keys.iter().flatten().unique())
```

Comment:
`.unique` looks unnecessary here?

Reply:
IMO It's not a big problem to have or not to have `.unique`. But I slightly prefer having it here.

Reply:
it's unnecessary performance overhead

Reply:
But when there're duplicated keys, the I/O path and cache `put` are also performance overhead.

Reply:
TBH I don't think there's visible difference on the overall throughput of an average query that contains `StreamAppendOnlyDedup`. I'm just doing it for mental cleanness. But I can do a bench if you want.

Reply:
Hmmm, I mean in `populate_cache` there's `self.cache.contains` check. Why there will be IO in this case?